queries/c/pairs.scm

```diff
@@ -0,0 +1,43 @@
+(preproc_if
+  "#if"    @left
+  "#endif" @right)
+
+(preproc_ifdef
+  "#ifdef" @left
+  "#endif" @right)
+
+(preproc_ifdef
+  "#ifndef" @left
+  "#endif" @right)
+
+(preproc_defined
```

Comment:
Couldn't this just be a `(_ "(" @left ")" @right)`. Otherwise we might have the risk that we forgot one node. But we can leave it that way since you already did the work.

Reply:
oh, I didn't know you can do that :laughing:
I thought it needs to target a specific node and find `()`, `{}` etc inside them.

Reply:
`(_ "(" @left ")" @right)` is also a lot faster to copy to every language

Reply:
we should probably do that for *all* pairs including `{` `(`, etc, but that's for someone else to fix :laughing:

Reply:
have one base.scm from which almost languages inherit

Reply:
Shouldn't have this published before fixing that :disappointed:

Reply:
I'll fix it tomorrow if no one does that before me, it'll be easier to support almost *all* languages

Reply:
The CI job could also add `; inherits base` for pairs.scm that don't exist yet.

Reply:
sounds like a good idea, then we'd only need to add some language specific pairs like `#if`, `#endif`, etc